packages/union-component/src/components/DonationPaymentForm.tsx

```diff
@@ -98,6 +109,11 @@ const DonationPaymentForm: React.FC<Props> = ({
     setFormData((state) => ({ ...state, phoneNumber }));
   };
 
+  const setFundId = useCallback(
+    (fundId: number) => setFormData((state) => ({ ...state, fundId })),
```

Comment:
I feel we pass throughout this already. Why not to use `onChangeInput`? why do we need a different state of the fund id?

Reply:
@duranmla this is needed due to I use this function to set the first fund when the API provides a response

`DonationFundDropdown.tsx`
```tsx
useEffect(() => {
    const getOptions = async () => {
      const funds = await getDonationFunds();

      if (funds?.length) {
        setOptions(funds);
        setFundId(Number(funds[0].id));
      }
    };

    getOptions();
  }, [setFundId]);
```

Reply:
Got it, in the end you are using `FormData` but you need to provide the initial value whenever the data has loaded. What I see here is that as you follow the naming convention of `setFundId` my expectations is that the method will be used every time a fund id is selected _(which is not true, and while that its ok it is confusing)_

I would change the method instead `setFundId` to `onFundsLoaded`
_With that I think we are good to go!_